versioned_hdf5/subchunk_map.py

```diff
@@ -135,13 +136,13 @@ def __init__(
         dset_size: hsize_t,
         chunk_size: hsize_t,
     ):
+        if idx.step <= 0:
+            raise NotImplementedError("Slice step must be positive")
+
         self.start = idx.start
         self.stop = idx.stop
         self.step = idx.step
 
-        if self.step <= 0:
-            raise NotImplementedError(f"Slice step must be positive not {self.step}")
```

Comment:
Unreachable: because self.step is unsigned, it would crash at line 140 with an opaque OverflowError